### ISSUE TITLE
Make tool bar size fixed regardless of screen resolution

### ISSUE THREAD

**Issue by esnosy** [Bforartists/Bforartists#3979]
while working on #3971 and comparing `space_toolbar.cc` with `space_topbar.cc`, I noticed that we can try to make the toolbar size fixed,
in the past we resorted to resaving the default blend with the window maximized at 1920x1080 resolution, but this isn't really a solution,
the size is still messed up on computers with different resolutions, top bar doesn't suffer from this, so maybe we can copy some of the code

**Comment by ReinerBforartists** [comment]
I am green with every improvement :)

**Comment by ReinerBforartists** [comment]
Related Cleanup space_toolbar.cc #3971

**Comment by Blenux** [comment]
How about a global one? Been on and off with this since october last year, was not 100% happy with it's pros and cons,

Pro = Access anytime from a popup to toggle any item.
Con = Can only have the one etc.

https://github.com/Bforartists/Bforartists/assets/25260650/f85e3bab-cc37-4494-8460-9cd896e45835

**Comment by iyadahmed** [comment]
The current one is great, and similar to Autodesk apps and SoftImage I
suppose, we should only fix the size problem

**Comment by iyadahmed** [comment]
I agree, at least we shouldn't delete the current one ^^

**Comment by Draise14** [comment]
"Why not both?"

Toolbar editor (opt in) + Toggeable Fixed toolbar (opt-out)

New default: only Toggeable Fixed toolbar

**Comment by ReinerBforartists** [comment]
Do we have a defined design decision yet? I think this is the bigger problem at the moment. That everybody wants it to do his own way. This should be fixed first before we make it a task.

**Comment by iyadahmed** [comment]
> Do we have a defined design decision yet? I think this is the bigger problem at the moment. That everybody wants it to do his own way. This should be fixed/defined first before we make it a task.

I think it is fine issue is the same, Blenux fix can close this issue too ^^

**Comment by Blenux** [comment]
Ok will do, just need to clean up some code and find some time to do it all (weekend most likely).

**Comment by Blenux** [comment]
Still on this, just been busy.

**Comment by Blenux** [comment]
[Test Branch is up](https://github.com/Bforartists/Bforartists/tree/blenux_topbar)

**Comment by Draise14** [comment]
Awesome Possum. Will test today.

**Comment by Draise14** [comment]
Tested @Blenux !! Looks super nice, and will avoid people loading up collapsed or expanded topbars in the future, can't wait! Works. 

## Observations
- Some of the sub-group options are still incomplete, ei, you toggle and they don't toggle. But as a prototype, nice. I assume it's polishing it and viola. 
https://github.com/Bforartists/Bforartists/assets/35357409/ce84a14a-bd91-4132-95ea-13d4420a955b
- The toolbar editor as secondary is all good, but now the options of the toolbar editor would be inconsistent
- The new notices in the drop downs are kinda useful, though maybe tooltips could help also/alternatively? For some reason the labels are red. 
![image](https://github.com/Bforartists/Bforartists/assets/35357409/c6b0c849-83ee-45c5-a5ac-2658ba578fd8)
- I think the reset panel options is interesting. Might be worth keeping, but would need good documentation. 
- The the toggle visibility in the options panel, not sure it's a good ideas, as you turn it off, then you can't access that panel again when clicking off, a user might get lost unless they know where to toggle it in the first place. 
- Button and background might need some work, button with same colour as header makes it feel like only icons vs buttons. 
- The sub-group drop down toggles could be the same style as the options panel. 
- Having the sub-group toggles in the options panel, I get the intention but feels like a different UX - though it's a short-hand method to the same. It might be redundant or double entry. I think I talked of this with Reiner a while back with a similar proposal, since it's a sub-group option then located in two places... the proposal was opted out to have less options bloat and categorized to the sub-group as a customization UX (1. set group. 2. Customize each group accordingly), which is more "assigned" to your workflow, with options that are less disconnect by keeping the sub-options in the sub-group. You most often toggle these not often, then save startup. If you have quick toggles on, then you can show and collapse and customize from a top level well enough. - the need for an sub-groups options short hand in the options panel to toggle groups properties often, and still one at a time - then you'd be doing more work ultimately, same thing with some draw backs. 
- Missing feature, quick toggles
![image](https://github.com/Bforartists/Bforartists/assets/35357409/5125634b-9e79-4ead-9762-0e8ba17debe5)
- I like the buttons over the tick boxes to a degree, bigger "hit" area, though they are the same. The rest could be the same for the sub-group options too and also be consistent. 


**So.. tasks:**
1. Make the options panel for the topbar and editor consistent with each other
2. Make it feature complete
3. Make the toggle in the window header say "Show Topbar" for consistency
4. Make background same as toolbar editor
5. Make options and sub-group options consistent between both
6. Remove the visibility toggle from the options panel
7. Add feature for "show all" quick toggles like the original editor
8. Remove double entries to toggle sub-group options

Nice work though. Will have to poke into the code to see how to help work this!

**Comment by Blenux** [comment]
Yeah still need to work on it, has I lost some of the work had to start from scracth bah, will work on it over the weekend last day of work today, so have some free time.

**Comment by Blenux** [comment]
Gonna try to get this topbar sorted asap.

**Comment by Draise14** [comment]
Really looking forward to it!! I wonder how many users have missed the toolbar due to smaller screen resolutions (it collapses)

**Comment by Blenux** [comment]
[updated topbar branch](https://github.com/Bforartists/Bforartists/tree/blenux_topbar)

**Comment by Draise14** [comment]
Will compile and test after breakfast

**Comment by iyadahmed** [comment]
I really like it it is amazing!
![image](https://github.com/Bforartists/Bforartists/assets/25552173/184f6783-1708-486a-aeea-ee872dd5d350)

**Comment by ReinerBforartists** [comment]
Yeah this is cool :)
We need to fix the layouts then. One bar should be enough ...

**Comment by Blenux** [comment]
[Updated Topbar Branch](https://github.com/Bforartists/Bforartists/tree/blenux_topbar), can push to merge if all is satisfied with it.

![image](https://github.com/Bforartists/Bforartists/assets/25260650/437ef092-74ee-44e0-b0eb-85d4accd80c3)

**Comment by Draise14** [comment]
Compiling to test!

**Comment by Draise14** [comment]
![image](https://github.com/Bforartists/Bforartists/assets/35357409/6b4c8eb3-658f-470f-b16a-792576b73016)

Nice, looks like we got some double registries of a class, but other than that it looks fine.

The principle options are now different in the top bar, so maybe consistency with the editor one way to the other could be nice. New one is nice. Could be another task to port it over to the editor. I'll poke around... 

Might change this to "topbar" instead of Bforaritsts topbar. We already in Bforartists. 
![image](https://github.com/Bforartists/Bforartists/assets/35357409/2ada1257-9606-4d4f-b998-0914bbe9b1d5)

**Comment by Blenux** [comment]
Interesting, they should be already fixed (did a few commits after work), except for the double registries of a classes, forgot about that one.

![image](https://github.com/Bforartists/Bforartists/assets/25260650/f6bb7a50-c89b-49c7-a386-4a0342ae20bc)

**Comment by Draise14** [comment]
Will test again, and hopefully get this into 4.1

**Comment by Draise14** [comment]
Ok, some minor labeling things to fix up, quick documentation update... and try make the editor consistent with the new system.

**Comment by Draise14** [comment]
![image](https://github.com/Bforartists/Bforartists/assets/35357409/1c19f55a-97f7-4eaf-9b0a-336e5f37805e)

Also, gottta add the quick toggles feature to the top bar.

**Comment by Draise14** [comment]
Ah, forgot to pull. Spent 5 hours making the old system toggle and it was awesome.. then I pulled and had merge conflicts and now I've got it all ruined...

**Comment by Blenux** [comment]
Yep can do that today.

**Comment by Draise14** [comment]
For the record, I do like your new configuration panel a lot more, and I had it also make a conditional so the quick toggles only show on the group you have set, which I found a lot more user friendly. I was trying to recover that after a botched merge... and wondering how to back migrate the same panels to the old toolbar.

But with the forgotten pull, I really screwed it up. I'm high strung these days, making me do big oops.

**Comment by Blenux** [comment]
Which configuration panel?

**Comment by Draise14** [comment]
The new one you had built, it is.. nice, not gonna lie! We can make a new task later after I merge this in.

**Comment by Draise14** [comment]
Ok, recovered the work I did yesterday, will commit and merge to master, and document. Will do it in another branch just in case.

**Comment by Draise14** [comment]
Waiting on merge to merge to master. Will document now.

**Comment by Draise14** [comment]
Gonna assign you @Blenux to this one so I don't forget the nice juicy bounty for this one.
